packages/opentelemetry/src/spanExporter.ts

```diff
@@ -159,26 +119,23 @@ export class SentrySpanExporter {
 
     // If the span doesn't have a local parent ID (it's a root span), we're gonna flush all the ended spans
     const localParentId = getLocalParentId(span);
-    if (!localParentId || this.isSpanAlreadySent(localParentId)) {
-      this._clearTimeout();
-
-      // If we got a parent span, we try to send the span tree
-      // Wait a tick for this, to ensure we avoid race conditions
-      this._flushTimeout = setTimeout(() => {
-        this.flush();
-      }, 1);
+    if (!localParentId || this._sentSpans.has(localParentId)) {
+      this._debouncedFlush();
     }
   }
 
   /** Try to flush any pending spans immediately. */
   public flush(): void {
-    this._clearTimeout();
-
-    const finishedSpans: ReadableSpan[] = this._finishedSpanBuckets.flatMap(bucket =>
-      bucket ? Array.from(bucket.spans) : [],
-    );
+    const finishedSpans: ReadableSpan[] = [];
+    for (const bucket of this._finishedSpanBuckets) {
+      if (bucket) {
+        for (const span of bucket.spans) {
+          finishedSpans.push(span);
+        }
+      }
+    }
```

Comment:
Is this relly chaeper than what we had? As far as I'm aware `.map()`/`.flatMap()` is always faster than building an array on the fly due to resizing and the interpreter not knowing the type of the items ahead of time.

Also, what's the expected amount of items here as if it is a handful we might be bikeshedding.

Reply:
buckets can be up to 1000 items large (which also does happen in reality). So not massive but also not tiny, basically :) 

It is still always creating a new array (`Array.from()`), I think map/flatMap themselves are fine! 🤔